### PR TITLE
Feature/dashboard

### PR DESCRIPTION
**Pull request by alexsicart**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,23 +20,34 @@
   },
   "homepage": "https://github.com/jmporchet/bravakin-client#readme",
   "dependencies": {
-    "bootstrap": "^4.0.0-alpha.6",
+    "@kadira/storybook": "^2.35.3",
+    "axios": "^0.16.2",
+    "bootstrap": "^4.0.0-beta",
+    "faker": "^4.1.0",
```
Comment (jmporchet) [inline package.json]: do we sill need faker?

File: package.json [hunk 2]
```diff
@@ -20,23 +20,34 @@
...
     "gulp": "^3.9.1",
     "gulp-eslint": "^4.0.0",
     "gulp-load-plugins": "^1.5.0",
     "jquery": "^3.2.1",
+    "json-schema-faker": "^0.4.3",
```
Comment (jmporchet) [inline package.json]: faker?

File: package.json [hunk 3]
```diff
@@ -20,23 +20,34 @@
...
+    "object-assign": "^4.1.1",
```
Comment (jmporchet) [inline package.json]: Where is this package used?

File: public/style.css [hunk 4]
```diff
@@ -0,0 +1,2 @@
+body, html {
```
Comment (jmporchet) [inline public/style.css]: Is this file really needed?

File: src/components/Dashboard/card.js [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+import { connect } from 'react-redux';
+
+
+import styles from './card.css';
+import FakeData from './fakeData.js';
+import KeyHandler, {KEYPRESS} from 'react-key-handler';
+
+
+
+class Card2 extends Component {
```
Comment (jmporchet) [inline src/components/Dashboard/card.js]: Use a more descriptive class name like InstagramCard or LikeCard

File: src/components/Dashboard/card.js [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+
+    this.fetchLikeableMedia()
+  }
+
+  fetchLikeableMedia () {
+    this.setState({fetching: true});
+    fetch(`http://192.168.0.49:3000/tags/${this.props.hashtags[this.state.hashtagIndex]}`, {
+      method: "GET",
+      headers: {
+        'Authorization': 'Bearer 5885499160.38553e7.ccfd98b2185a4fed833163bf17e86b04',
```
Comment (jmporchet) [inline src/components/Dashboard/card.js]: Don't hardcode the token

``` 'Authorization': 'Bearer ' + this.props.access_token```

File: src/components/Dashboard/card.js [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+        const card = this.state.stack.getCard(el);
+        card.throwOut(100, 200, direction);
+      }
+
+
+      const currentIndex = this.state.currentIndex - 1;
+      this.setState({currentIndex: currentIndex});
+
+      if (currentIndex < this.state.floor) {
+        // TODO: call `this.fetchLikeableMedia()`
```
Comment (jmporchet) [inline src/components/Dashboard/card.js]: remove comment

File: src/components/Dashboard/card.js [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+        this.fetchLikeableMedia();
+      }
+    }
+  }
+
+
+  onThrowOut = (e)=> {
+    // TODO: post like
```
Comment (jmporchet) [inline src/components/Dashboard/card.js]: remove comment

File: src/components/Dashboard/card.js [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+    const cardId = e.target.getAttribute('id').split('card')[1];
+    const media = this.state.cards[cardId];
+    console.log(media);
+
+    if(e.throwDirection === Direction.LEFT) {
+      console.log('Don\'t like it.');
```
Comment (jmporchet) [inline src/components/Dashboard/card.js]: change this

File: src/components/Dashboard/card.js [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+    else {
+      console.log('I like it.');
+      fetch(`http://192.168.0.49:3000/media/like`, {
+        method: 'POST',
+        body: JSON.stringify({
+          url: media.url
+        }),
+        headers: {
+          'Content-Type': 'application/json',
+          'Authorization': 'Bearer 5885499160.38553e7.ccfd98b2185a4fed833163bf17e86b04'
```
Comment (jmporchet) [inline src/components/Dashboard/card.js]: token

File: src/private/conf.js [hunk 11]
```diff
@@ -0,0 +1,4 @@
+export default {
```
Comment (jmporchet) [inline src/private/conf.js]: This file should have been .gitignored
Don't expose these secrets on Github

File: src/components/Dashboard/fakeData.js [hunk 12]
```diff
@@ -0,0 +1,35 @@
+import faker from 'faker';
```
Comment (jmporchet) [inline src/components/Dashboard/fakeData.js]: Do we still need this file?

File: package.json (hunk 1 above)
Comment (alexsicart) [reply package.json]: Done

File: package.json (hunk 2 above)
Comment (alexsicart) [reply package.json]: Done

File: package.json (hunk 3 above)
Comment (alexsicart) [reply package.json]: Object assign it is in use, json schema not

File: public/style.css (hunk 4 above)
Comment (alexsicart) [reply public/style.css]: Moved to trash!

File: src/components/Dashboard/card.js (hunk 5 above)
Comment (alexsicart) [reply src/components/Dashboard/card.js]: Changed to InstagramCard

File: src/components/Dashboard/card.js (hunk 6 above)
Comment (alexsicart) [reply src/components/Dashboard/card.js]: done

File: src/components/Dashboard/card.js (hunk 7 above)
Comment (alexsicart) [reply src/components/Dashboard/card.js]: done

File: src/components/Dashboard/card.js (hunk 8 above)
Comment (alexsicart) [reply src/components/Dashboard/card.js]: done

File: src/components/Dashboard/card.js (hunk 9 above)
Comment (alexsicart) [reply src/components/Dashboard/card.js]: Changed

File: src/components/Dashboard/card.js (hunk 10 above)
Comment (alexsicart) [reply src/components/Dashboard/card.js]: Changed

File: src/components/Dashboard/fakeData.js (hunk 12 above)
Comment (alexsicart) [reply src/components/Dashboard/fakeData.js]: Deleted

**Comment by alexsicart** [comment]
The app at this moment is working correctly, without any conflict.
Changes made:
- Installing the correct version of "react-key-handler", it didn't let the app work correctly.
- Deleting package-lock and merged package-lock.json from develop
- Minor big fixes that made the app more stressful.

Waiting for a review from @arol & @jmporchet 

Looking forward this changes be good.


AS

File: package.json [hunk 13]
```diff
@@ -4,7 +4,7 @@
   "description": "Bravakin client",
   "main": "index.js",
   "scripts": {
-    "start": "PORT=3001 react-scripts start",
+    "start": "PORT=3000 react-scripts start",
```
Comment (arol) [inline package.json]: Come back to 3001, we want to run this at the same time as the back-end

**Review by arol — DISMISSED** [review]

File: package.json (hunk 13 above)
Comment (alexsicart) [reply package.json]: Completed @arol Thanks!

**Comment by alexsicart** [comment]
All conflicts solve, and requested changes completed.

**Comment by alexsicart** [comment]
The last request for @arol was done, and the merge branch 'develop' into this branch was done.

AS

**Review by jmporchet — APPROVED** [review]

**Pull request merged**